kamu-cli/src/app.rs

```diff
@@ -63,8 +67,10 @@ pub async fn run(
 
     let mut command: Box<dyn Command> = cli_commands::get_command(&catalog, matches)?;
 
-    let result = if command.needs_workspace() && !in_workspace(catalog.get_one().unwrap()) {
+    let result = if command.needs_workspace() && !workspace_svc.is_in_workspace() {
         Err(CLIError::usage_error_from(NotInWorkspace))
+    } else if command.needs_workspace() && workspace_svc.is_upgrade_needed()? {
+        Err(CLIError::usage_error_from(WorkspaceUpgradeRequired))
```

Comment:
I wonder if a handling like this is necessary anywhere in the API server, if initialized with local workspace repository

Reply:
For simplicity let's assume for now we will always run API server on compatible version of local workspace.